src/core/include/openvino/op/rms_norm.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/op.hpp"
+
+namespace ov {
+namespace op {
+namespace v14 {
+/// \brief Operator performing Root Mean Square Normalization
+/// \ingroup ov_ops_cpp_api
+class OPENVINO_API RMSNorm : public ov::op::Op {
+public:
+    OPENVINO_OP("RMSNorm", "opset14", ov::op::Op);
+
+    RMSNorm() = default;
+    /// \brief Constructs an RMSNorm operation without scaling.
+    ///
+    /// \param data Input tensor with data
+    /// \param axes Axes for reduce mean calculation
+    /// \param eps Epsilon for not dividing by zero while normalizing the value
+    /// \param compute_type Precision for the internal computation, if undefined it's the same as the input type
```

Comment:
Maybe change it to if undefined then data input type will be used. It suggest will have same value as type of data input but it will not be changed.

Reply:
Ok, I will update this comment (within this PR or as a follow up).